app/Repositories/CacheUserRepository.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Gladiator\Repositories;
+
+use Gladiator\Models\User;
+use Illuminate\Support\Facades\Cache;
+
+class CacheUserRepository implements UserRepositoryInterface
+{
+    /**
+     * UserRepositoryInterface instance.
+     *
+     * @var \Gladiator\Repositories\UserRepositoryInterface
+     */
+    protected $database;
+
+    /**
+     * Create new CacheUserRepository instance.
+     *
+     * @param UserRepositoryInterface $database
+     */
+    public function __construct(UserRepositoryInterface $database)
+    {
+        $this->database = $database;
+    }
+
+    /**
+     * Create and cache a new user.
+     *
+     * @param  object  $account
+     * @return object
+     */
+    public function create($account)
+    {
+        $user = $this->database->create($account);
+
+        $this->resolveUpdatedRoles($user->id, $user->role);
+
+        $user = $this->find($user->id);
+
+        return $user;
+    }
+
+    /**
+     * Find the specified resource in cache or default to database lookup.
+     *
+     * @param  string  $id  Northstar ID
+     * @return object
+     */
+    public function find($id)
+    {
+        $user = Cache::get($id);
+
+        if (! $user) {
+            $user = $this->database->find($id);
+
+            $this->store($user->id, $user);
+        }
+
+        return $user;
+    }
+
+    /**
+     * Get collection of users from cache by specified role or default to database lookup.
+     *
+     * @param  string|null $role
+     * @return \Illuminate\Support\Collection
+     */
+    public function getAllByRole($role = null)
+    {
+        $key = array_search($role, User::getRoles());
```

Comment:
@DFurnes made the `$role` default to `null`, but the role of null is the key of `contestant` in regards to caching and when using gui things, so used this approach instead for grabbing the associated `key` for the `null` role ;)